mailpoet/tests/integration/Segments/DynamicSegments/Filters/WooCommerceNumberOfOrdersTest.php

```diff
@@ -142,4 +157,19 @@ private function cleanUp() {
     $this->connection->executeQuery("TRUNCATE TABLE {$wpdb->prefix}wc_customer_lookup");
     $this->connection->executeQuery("TRUNCATE TABLE {$wpdb->prefix}wc_order_stats");
   }
+
+  private function createSubscriber(): SubscriberEntity {
```

Comment:
You don't need this logic. We have [a subscriber factory](https://github.com/mailpoet/mailpoet/blob/master/mailpoet/tests/DataFactories/Subscriber.php) that should cover your requirements. If some functionality is missing, don't hesitate to extend it. [Here](https://github.com/mailpoet/mailpoet/blob/master/mailpoet/tests/integration/API/JSON/v1/SendingTaskSubscribersTest.php#L61-L65) is an example of usage.

The factory is mostly used in acceptance tests, but I think that we want to avoid the code repetition as here.